In [36]:
import scipy as sp
import scipy.stats
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

1. 어느 대학교에서 대중교통을 이용하여 등교하는 학생의 비율을 알아보기 위하여 이 학교 학생 중 100 명을 임의 추출하여 조사한 결과 50%의 학생이 대중교통을 이용하여 등교하는 것으로 나타났다. 이 결과를 이용하여 이 대학교 전체 학생 중에서 대중교통을 이용하여 등교하는 학생의 비율 p에 대한 신뢰도 95%의 신뢰 구간을 구하시오.

In [9]:
phat = 0.5
alpha = 0.05
n = 100

In [12]:
rv = sp.stats.norm

In [17]:
z = rv.ppf((1-alpha/2),loc=0,scale=1)

In [18]:
z

1.959963984540054

In [23]:
ll = phat - z*(phat*(1-phat)/n)**0.5
ul = phat + z*(phat*(1-phat)/n)**0.5

In [26]:
print(ll,ul)

0.4020018007729973 0.5979981992270027


2. 어느 음식점에서 새로운 메뉴를 개발하여 이 메뉴에 대한 선호도를 조사하기로 하였다. 고객 100명을 임의추출하여 이 메뉴에 대한 반응을 조사하였더니 이들 중 4/5가 선호한다고 하였다. 전체 고객의 새로운 메뉴에 대한 선호도를 p라 할 때, 모비율 p에 대한 신뢰도 95%의 신뢰구간을 구하시오.

In [27]:
n = 100
phat = 4/5
alpha = 0.05

In [28]:
ll = phat - z*(phat*(1-phat)/n)**0.5
ul = phat + z*(phat*(1-phat)/n)**0.5

In [29]:
print(ll,ul)

0.7216014406183979 0.8783985593816022


3. 우리나라 성인 남성의 흡연율을 조사한다고 한다. 이에 성인 남자 1,000명을 무작위로 뽑아 흡연 여부를 조사하였더니, 430명이 흡연을 하고 있었다. 이때 흡연율(모비율)에 대한 90% 신뢰구간을 추정하시오.

In [30]:
n = 1000
phat = 430/1000
alpha = 0.1

In [31]:
z = rv.ppf((1-alpha/2),loc=0,scale=1)

In [32]:
z

1.6448536269514722

In [34]:
ll = phat - z*(phat*(1-phat)/n)**0.5
ul = phat + z*(phat*(1-phat)/n)**0.5
print(ll,ul)

0.40424871458356326 0.4557512854164367


4. A회사의 건전지의 수명시간이 1000시간 일 때, 무작위로 뽑은 10개의 건전지에 대한 수명은 다음과 같다.
980, 1008, 968, 1032, 1012, 996, 1021, 1002, 996, 1017
샘플이 모집단과 같다고 할 수 있는가?

Hypothesis  
H(0) : 건전지의 수명 = 1000 시간   
H(1) : 건전지의 수명 != 1000시간

In [39]:
life = 1000
sample = np.array([980, 1008, 968, 1032, 1012, 996, 1021, 1002, 996, 1017])

In [160]:
np.mean(sample)

1003.2

In [148]:
#scipy.stats.shapiro(x)
i,v = sp.stats.shapiro(sample)
print('W :',i,',','P-value :',v)


W : 0.9757134914398193 , P-value : 0.938188374042511


유의 수준을 0.05로 하였을 때, 건전지의 수명은 정규분포를 따른다고 할수 있음

In [157]:
result = sp.stats.ttest_1samp(sample, popmean=1000)

In [159]:
print("The T-statistic is %.3f and the p-value is %.3f" % result)

The T-statistic is 0.527 and the p-value is 0.611


검정결과 p-value 값이 0.611이므로 유의수준 0.05 하에서 귀무가설을 기각하지 못한다.
즉 추출한 샘플이 모집단과 같다고 결론내린다.

5. 어떤 반의 학생들의 수학 평균성적은 55점이었다. 0교시 수업을 시행하고 나서 학생들의 시험 성적은 다음과 같다.
58, 49, 39, 99, 32, 88, 62, 30, 55, 65, 44, 55, 57, 53, 88, 42, 39
0교시 수업을 시행한 후, 학생들의 성적은 올랐다고 할 수 있는가?